node/src/main/scala/coop/rchain/node/encode/JsonEncoder.scala

```diff
@@ -78,6 +86,104 @@ object JsonEncoder {
+  implicit val decodeEMinusMinus: Decoder[EMinusMinus]         = deriveDecoder[EMinusMinus]
+  implicit val decodeEMod: Decoder[EMod]                       = deriveDecoder[EMod]
+  implicit val decodeExprInstance: Decoder[ExprInstance]       = deriveDecoder[ExprInstance]
+  implicit val decodeExpr: Decoder[Expr]                       = deriveDecoder[Expr]
+  implicit val decodeMatchCase: Decoder[MatchCase]             = deriveDecoder[MatchCase]
+  implicit val decodeMatch: Decoder[Match]                     = deriveDecoder[Match]
+  implicit val decodeGPrivate: Decoder[GPrivate]               = deriveDecoder[GPrivate]
+  implicit val decodeGDeployId: Decoder[GDeployId]             = deriveDecoder[GDeployId]
+  implicit val decodeGDeployerId: Decoder[GDeployerId]         = deriveDecoder[GDeployerId]
+  implicit val decodeGSysAuthToken: Decoder[GSysAuthToken]     = deriveDecoder[GSysAuthToken]
+  implicit val decodeUnfInstance: Decoder[UnfInstance]         = deriveDecoder[UnfInstance]
+  implicit val decodeGUnforgeable: Decoder[GUnforgeable]       = deriveDecoder[GUnforgeable]
+  implicit val decodeBundle: Decoder[Bundle]                   = deriveDecoder[Bundle]
+  implicit val decodeVarRef: Decoder[VarRef]                   = deriveDecoder[VarRef]
+  implicit val decodeConnectiveBody: Decoder[ConnectiveBody]   = deriveDecoder[ConnectiveBody]
+  implicit val decodeConnectiveInstance: Decoder[ConnectiveInstance] =
+    deriveDecoder[ConnectiveInstance]
+  implicit val decodeConnective: Decoder[Connective] = deriveDecoder[Connective]
+  implicit val decodeAlwaysEqual: Decoder[AlwaysEqual[BitSet]] =
+    Decoder.decodeUnit.map[AlwaysEqual[BitSet]](_ => AlwaysEqual(BitSet()))
+
+  // FIXME blake2b512Random decode. Question Is that really neccessary?
+  implicit val decodeBlake2b512Random: Decoder[Blake2b512Random] =
+    Decoder.decodeUnit.map[Blake2b512Random](_ => Blake2b512Random(1))
+
+  // convert the circe codec into scodec
+  def convertCcodecToScodec[A](encoder: Encoder[A], decoder: Decoder[A]): SCodec[A] =
+    new SCodec[A] {
+      override def encode(value: A): Attempt[BitVector] = utf8.encode(encoder.apply(value).noSpaces)
+
+      override def sizeBound: SizeBound = unknown
+
+      override def decode(bits: BitVector): Attempt[DecodeResult[A]] =
+        utf8
+          .decode(bits)
+          .flatMap(
+            s =>
+              Attempt
+                .fromEither(
+                  parse(s.value)
+                    .leftMap(
+                      e => Err.General(e.message, e.underlying.getStackTrace.map(_.toString).toList)
+                    )
+                    .map(j => decoder.decodeJson(j))
+                )
+                .flatMap(
+                  r =>
+                    Attempt.fromEither(
+                      r.leftMap(d => Err.General(d.message, List.empty))
+                        .map(DecodeResult(_, BitVector.empty))
+                    )
+                )
+          )
+    }
```

Comment:
I'd suggest we take some time to make code more compact, e.g. this can be rewritten as, and imo much more easy to understand
```scala
override def decode(bits: BitVector): Attempt[DecodeResult[A]] =
        for {
          string <- utf8.decode(bits).map(_.value)
          json <- Attempt.fromEither(parse(string).leftMap { e =>
                   Err.General(e.message, e.underlying.getStackTrace.map(_.toString).toList)
                 })
          result <- Attempt.fromEither(
                      decoder
                        .decodeJson(json)
                        .map(DecodeResult(_, BitVector.empty))
                        .leftMap(d => Err.General(d.message, List.empty))
                    )
        } yield result
```

Reply:
https://github.com/rchain/rchain/pull/3454#discussion_r660026071
@tgrospic I am not sure if this is applicable here. But I agree with you.